File: Python/448.py
# 1
class Solution:
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        return list(set(range(1, len(nums) + 1)) - set(nums))

        
# 2
class Solution:
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        appeared = [0] * len(nums)
        for n in nums:
            appeared[n - 1] = 1
        ret = []
        for i in range(0,len(appeared)):
            if appeared[i] == 0:
                ret.append(i + 1)
        return ret

File: C++/448.cpp
// #1
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int> &nums) {
        auto size = nums.size();
        for (int i = 0; i < size; ++i)
            while (nums[i] != i + 1 && nums[i] != nums[nums[i] - 1])
                swap(nums[i], nums[nums[i] - 1]);
        vector<int> ret;
        for (int i = 0; i < size; ++i)
            if (nums[i] != i + 1)
                ret.push_back(i + 1);
        return ret;
    }
};


// #2
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int> &nums) {
        auto size = nums.size();
        vector<int> appeared(size, 0), ret;
        for (auto n :nums)
            appeared[n - 1] = 1;
        for (int i = 0; i < size; ++i)
            if (appeared[i] == 0)
                ret.push_back(i + 1);
        return ret;
    }
};